Topics/Tech_Stacks/Swift_Mapkit.md

```diff
@@ -0,0 +1,263 @@
+  import MapKit
+    
+  struct MapContentView: View {
+     let robarts = CLLocationCoordinate2D(latitude: 43.664486, longitude: -79.399689)
+    var body: some View {
+             Map(){
+                    Marker("Robarts", coordinate: robarts)
+                 }
+     }
+  }
+   
+  #Preview {
+       MapContentView()
+  }
+
+  ```
+
+
+### Marker Customization
+There are a couple of ways to customize your marker, such as color and what is displayed in the marker bubble.
+<p align="center">
+<img width=1000 src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/113125436/6d12cc84-1e81-4038-a3d0-cac549de6a44"/>
+</p>
+
+**Color**
+  
+  To change color of your marker you can apply the `.tint()` method. Tint takes in a Color objext so you can either do a systen color such as `.blue` or your own custom color.
+
+  ```swift
+     Marker("Robarts", coordinate: robarts).tint(.blue)
+  ```
+
+**Icon**
+
+  To change the what is displayed in the Marker bubble you can add the `systemImage`, `image`, or `monogram` parameters to your Marker as shown below.
+
+  System Image:
+  ```swift
+     Marker("Bahen", systemImage: "star.fill", coordinate: bahen)
+  ```
+   Image:
+   ```swift
+     Marker("Robarts", image: "<Your Image Name>", coordinate: robarts)
+   ```
+
+  Monogram
+  ```swift
+     Marker("University College", monogram: Text("UC"), coordinate: uc)
+  ```
+
+If you want more freedom with your designing map pins, try using [Annotations](#annotations) instead.
+## Annotations
+Annotations allow for MapKit developrs to completely redesign the location coordinate indicators.
+Unlike markers which had a set amount of parameters and there was no way to escape the bubble, Annotations instead takes in a Swift UI `View()`.
+
+For example, here we made the annotation to just be a system image with padding rather than a marker bubble.
+```swift
+import SwiftUI
+import MapKit
+
+struct MapContentView: View {
+    
+    let robarts = CLLocationCoordinate2D(latitude: 43.664486, longitude: -79.399689)
+    
+    var body: some View {
+        Map(){
+            Annotation("Robarts", coordinate: robarts){
+                Image(systemName: "books.vertical")
+                    .padding(6)
+                    .foregroundColor(.white)
+                    .background(/*@START_MENU_TOKEN@*/.blue/*@END_MENU_TOKEN@*/)
+            }
+            
+        }
+    }
+}
+
+#Preview {
+    MapContentView()
+}
+```
+<p align="center">
+<img width=1000 src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/113125436/c64ff5e5-8286-4ad4-b24f-3eae8545f247"/>
+</p>
+
+Overall, the annotations allow you to make the map pins compeletely your own.
+
+## Camera Position
+By default the map focuses on the map contents (markers, annotaions, etc). To focus on a particular region or location, we can use `MapCameraPosition`. This can be helpful in a couple cases such as:
```

Comment:
```suggestion
By default the map focuses on the map contents (markers, annotations, etc). To focus on a particular region or location, we can use `MapCameraPosition`. This can be helpful in a couple cases such as:
```

Reply:
Wow I have so many typos